clang/lib/Sema/SemaChecking.cpp

```diff
@@ -2110,6 +2110,66 @@ static bool checkFPMathBuiltinElementType(Sema &S, SourceLocation Loc,
   return false;
 }
 
+/// SemaBuiltinCpuSupports - Handle __builtin_cpu_supports(char *).
+/// This checks that the target supports __builtin_cpu_supports and
+/// that the string argument is constant and valid.
+static bool SemaBuiltinCpuSupports(Sema &S, const TargetInfo &TI,
```

Comment:
most of the source code of function `SemaBuiltinCpuSupports` and  `SemaBuiltinCpuIs` are same , we can add helper function to reduce the code.